github/copilot.go

```diff
@@ -0,0 +1,284 @@
+// Copyright 2023 The go-github AUTHORS. All rights reserved.
+//
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package github
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+)
+
+// CopilotService provides access to the Copilot-related functions
+// in the GitHub API.
+//
+// GitHub API docs: https://docs.github.com/en/rest/copilot/
+type CopilotService service
+
+// CopilotOrganizationDetails represents the details of an organization's Copilot for Business subscription.
+type CopilotOrganizationDetails struct {
+	SeatBreakdown         *CopilotSeatBreakdown `json:"seat_breakdown"`
+	PublicCodeSuggestions string                `json:"public_code_suggestions"`
+	CopilotChat           string                `json:"copilot_chat"`
+	SeatManagementSetting string                `json:"seat_management_setting"`
+}
+
+// CopilotSeatBreakdown represents the breakdown of Copilot for Business seats for the organization.
+type CopilotSeatBreakdown struct {
+	Total               int `json:"total"`
+	AddedThisCycle      int `json:"added_this_cycle"`
+	PendingCancellation int `json:"pending_cancellation"`
+	PendingInvitation   int `json:"pending_invitation"`
+	ActiveThisCycle     int `json:"active_this_cycle"`
+	InactiveThisCycle   int `json:"inactive_this_cycle"`
+}
+
+// ListCopilotSeatsResponse represents the Copilot for Business seat assignments for an organization.
+type ListCopilotSeatsResponse struct {
+	TotalSeats int64                 `json:"total_seats"`
+	Seats      []*CopilotSeatDetails `json:"seats"`
+}
+
+// CopilotSeatDetails represents the details of a Copilot for Business seat.
+type CopilotSeatDetails struct {
+	// Assignee can either be a User, Team, or Organization.
+	Assignee                interface{} `json:"assignee"`
+	AssigningTeam           *Team       `json:"assigning_team,omitempty"`
+	PendingCancellationDate *string     `json:"pending_cancellation_date,omitempty"`
+	LastActivityAt          *Timestamp  `json:"last_activity_at,omitempty"`
+	LastActivityEditor      *string     `json:"last_activity_editor,omitempty"`
+	CreatedAt               *Timestamp  `json:"created_at"`
+	UpdatedAt               *Timestamp  `json:"updated_at,omitempty"`
+}
+
+// SeatAssignments represents the number of seats assigned.
+type SeatAssignments struct {
+	SeatsCreated int `json:"seats_created"`
+}
+
+// SeatCancellations represents the number of seats cancelled.
+type SeatCancellations struct {
+	SeatsCancelled int `json:"seats_cancelled"`
+}
+
+func (cp *CopilotSeatDetails) UnmarshalJSON(data []byte) error {
+	type alias CopilotSeatDetails
+	var seatDetail alias
```

Comment:
I'm not sure I understand the need for line 67 or why we can't have `var seatDetails CopilotSeatDetails`.
Can you please either add comments as to why these are necessary or just remove them if they are not needed?

Reply:
The `alias` type has the `CopilotSeatDetails` struct fields, but not its methods, this way when we call `json.Unmarshal` inside this function we don't have an infinite recursion. I'll add the comment and make the other requested changes tomorrow 🙂